sos/plugins/__init__.py

```diff
@@ -1256,27 +1257,99 @@ def collect_cmd_output(self, cmd, suggest_filename=None,
         )
 
     def exec_cmd(self, cmd, timeout=300, stderr=True, chroot=True, runat=None,
-                 env=None, binary=False, pred=None):
+                 env=None, binary=False, pred=None, container=False):
         """Execute a command right now and return the output and status, but
         do not save the output within the archive.
 
         Use this method in a plugin's setup() if command output is needed to
         build subsequent commands added to a report via add_cmd_output().
         """
+        _default = {'status': None, 'output': ''}
         if not self.test_predicate(cmd=True, pred=pred):
-            return {
-                'status': None,
-                'output': ''
-            }
+            return _default
 
         if chroot or self.commons['cmdlineopts'].chroot == 'always':
             root = self.sysroot
         else:
             root = None
 
+        if container:
+            if self.container_runtime is None:
+                self._log_info("Cannot run cmd '%s' in container %s: no "
+                               "runtime detected on host." % (cmd, container))
+                return _default
+            if self.container_exists(container):
+                cmd = self.fmt_container_cmd(container, cmd)
+            else:
+                self._log_info("Cannot run cmd '%s' in container %s: no such "
+                               "container is running." % (cmd, container))
+
         return sos_get_command_output(cmd, timeout=timeout, chroot=root,
                                       chdir=runat, binary=binary, env=env)
 
+    def container_exists(self, name):
+        """If a container runtime is present, check to see if a container with
+        a given name is currently running
+        """
+        if self.container_runtime:
+            con = self.container_runtime.get_container_by_name(name)
+            return con is not None
+        return False
+
+    def get_container_by_name(self, name):
+        if self.container_runtime:
+            return self.container_runtime.get_container_by_name(name)
+
+    def get_containers(self, runtime=None, get_all=False):
+        """Return a list of all container IDs from the Policy's
+        ContainerRuntime.
+
+        If `runtime` is provided, return an empty list if the Policy's loaded
+        ContainerRuntime matches that runtime.
```

Comment:
.. if the loaded ContainerRuntime does _not_ match that runtime..?

Reply:
Done.